cmake/CMakeLists.txt

```diff
@@ -81,24 +82,17 @@ set(protobuf_BUILD_TESTS OFF CACHE BOOL "Build protobuf tests" FORCE)
 set(NSYNC_ENABLE_TESTS OFF CACHE BOOL "Build protobuf tests" FORCE)
 set(ONNX_ML 1)
 
-if(onnxruntime_ENABLE_LTO)
-  #LTO(or LTCG) is great, in our case it can reduce binary size by 1/3.
-  #cmake can only help us check if the compiler support LTO or not, it can't tell us if the feature works well
-  #Don't enable this option in Ubuntu 16.04, protoc will crash
-  include(CheckIPOSupported)
-  check_ipo_supported(RESULT ipo_enabled OUTPUT ipo_output)
-  #TODO: figure out why nsync doesn't work
-  if(NOT onnxruntime_USE_NSYNC)
-    if(ipo_enabled)
-      set(CMAKE_INTERPROCEDURAL_OPTIMIZATION_RELEASE ON)
-      set(CMAKE_INTERPROCEDURAL_OPTIMIZATION_RELWITHDEBINFO ON)
-    else()
-      message(WARNING "IPO is not supported: ${ipo_output}")
-    endif()
-  endif()
+
+include(CheckIPOSupported)
+check_ipo_supported(RESULT ipo_enabled OUTPUT ipo_output)
```

Comment:
not sure about this logic.
we don't seem to do anything with ipo_enabled anymore. then why call check_ipo_supported?
the previous logic made more sense, where we do the check only if the cmake option is enabled.  (even though we default to ON here, someone could still set it to OFF)